torch/distributions/transforms.py

```diff
@@ -600,6 +601,26 @@ def log_abs_det_jacobian(self, x, y):
         return -F.softplus(-x) - F.softplus(x)
 
 
+class SoftplusTransform(Transform):
+    r"""
+    Transform via the mapping :math:`\text{Softplus}(x) = \log(1 + \exp(x))`.
+    """
+    domain = constraints.real_vector
+    codomain = constraints.positive
```

Comment:
```py
codomain = constraints.independent(constraints.positive, 1)
```
(this would have been caught by unit tests)

Reply:
What does `constraints.independent` do exactly? The test that I added (locally - I will upload soon) to `test/distributions/test_transforms.py` (adding a line `SoftplusTransform(cache_size=cache_size)` in `get_transforms`) will only pass if I specify this as `constraints.positive` (otherwise it gives an `AssertionError` at `assert torch.Size([4, 5]) == torch.Size([4])` in `test_jacobian(transform)`)

Reply:
Oh I'm sorry, I think this should actually assume `.event_dim = 0` via:
```py
domain = constraints.real  # not real_vector
codomain = constraints.positive
```
The `independent(-, 1)` and `real -> real_vector` both say to set `.event_dim = 1` which determines the shape of the `.log_abs_det_jacobian`. If `.event_dim == 0` then `.log_abs_det_jacobian()` is computed element-wise (which is what you want here). If `.event_dim == 1` then the jacobian is computed for the entire `shape[-1:]` in total, but you won't want that here since `SoftplusTransform` can apply to even scalar tensors (`.shape == ()`).

Reply:
Ah thank you - my bad for using `real_vector` :) Updated this now and added the test, thanks a lot for reviewing